central/complianceoperator/v2/report/manager/manager_impl.go

```diff
@@ -381,15 +385,29 @@ func (m *managerImpl) getScanConfigWatcher(ctx context.Context, results *watcher
 	if sc == nil {
 		return nil, nil, false, errors.Errorf("ScanConfiguration not found for scan %s", results.Scan.GetScanName())
 	}
-	var w watcher.ScanConfigWatcher
-	var watcherIsRunning bool
-	concurrency.WithLock(&m.watchingScanConfigsLock, func() {
-		if w, watcherIsRunning = m.watchingScanConfigs[sc.GetId()]; !watcherIsRunning {
-			log.Debugf("Staring config watcher %s", sc.GetId())
+	w, watcherIsRunning := concurrency.WithLock2[watcher.ScanConfigWatcher, bool](&m.watchingScanConfigsLock, func() (watcher.ScanConfigWatcher, bool) {
+		w, watcherIsRunning := m.watchingScanConfigs[sc.GetId()]
+		return w, watcherIsRunning
+	})
+	if !watcherIsRunning {
```

Comment:
Here starts a new behavior of this func that is no longer a good fit for its name. Could we at least make this behavior visible by renaming that to `getOrCreate...`

Reply:
Done